universal-application-tool-0.0.1/app/services/question/QuestionDefinition.java

```diff
@@ -0,0 +1,91 @@
+package services.question;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import com.google.common.collect.ImmutableMap;
+import java.util.Locale;
+
+/** Defines a single question. */
+public class QuestionDefinition {
+  private final String id;
+  private final String version;
+  private final String name;
+  private final String path;
+  private final String description;
+  private final ImmutableMap<Locale, String> questionText;
+  private final ImmutableMap<Locale, String> questionHelpText;
+
+  public QuestionDefinition(
+      String id,
+      String version,
+      String name,
+      String path,
+      String description,
+      ImmutableMap<Locale, String> questionText,
+      ImmutableMap<Locale, String> questionHelpText) {
+    this.id = checkNotNull(id);
+    this.version = checkNotNull(version);
+    this.name = checkNotNull(name);
+    this.path = checkNotNull(path);
+    this.description = checkNotNull(description);
+    this.questionText = checkNotNull(questionText);
+    this.questionHelpText = checkNotNull(questionHelpText);
+  }
+
+  /** Get the unique identifier for this question. */
+  public String getId() {
+    return this.id;
+  }
+
+  /** Get the system version this question is pinned to. */
+  public String getVersion() {
+    return this.version;
+  }
+
+  /** Get the name of this question. */
+  public String getName() {
+    return this.name;
+  }
+
+  /** Get the path of this question's parent, in JSON notation. */
```

Comment:
Suggestion (non-blocking): recommend including in comment that the question definition paths are what define the structure of an applicant's ApplicationData

Reply:
Question (non-blocking): is the path to the question's parent or to the question itself? I'd think the question itself e.g. for the applicant's name I'd expect the question definition path to be `applicant.name`.

Reply:
Maybe we discussed using the question name as the last segment of the path? If that's the case I'd advise against it because the name is meant to be human readable whereas the path should be optimized for machine-readability e.g. should not contain spaces and punctuation.

Reply:
Yeah, I think we were assuming that the name would be a valid camelCase string, but that may be a bit to code-y for admins so I agree that the name should be human readable in which case the path should be the full path instead of the path to the parent.

Plus, we can always get the parent given the full path - or we can just add a simple utility method.